### PR TITLE
feat(ch_ephem): CHIME ephemeris refactored from ch_util.

### PR DESCRIPTION
**Pull request by ketiltrout**
Extracts a lot of `ch_util.ephemeris` into a separate package.

## General changes:

* Solar and lunar ephemeris functions have been converted into `caput.time.Observer` methods: https://github.com/radiocosmology/caput/pull/271
* Other functions that take a `Observer` object, now use `None` as a default, and load the `chime` observer during execution if necessary.  (This is to avoid unnecessarily importing `ch_ephem.observers`.)
* Could use a test-suite, but this PR is big enough already.

More comments on specific submodules follow.

## Catalogs:

This contains all the `ch_util/catalogs` plus a function (`ch_ephem.catalogs.load`) to read them and return the parsed JSON.

The standard source catalogues loaded by the `sources` submodule need to move to `ch_ephem`.  The other catalogues could stay in `ch_util`, but it seems simpler to me to keep them all in the same place and provide a uniform interface to them.

## Coord:

Co-ordinate transforms etc.  These are essentially unchanged from their versions in `ch_util.ephemeris`.

## Observers:

Re-creates all the `caput.time.Observers` which used to be created by `ch_util.ephemeris`, but for the most part, they have to be fetched via name from the `get()` function.  The exception to this is the `chime` Observer, which can be directly imported, since it's the one usually needed.

Position data for the observers is now in a YAML data file `instruments.yaml` which is parsed at import time.  New instrument `Observer` objects can be added to `ch_ephem` just by adding information for it to the YAML file.

Also note that `Observers` returned by this submodule have an extra attribute, `rotation`, inserted into their `__dict__`, which is not in `caput`.  This rotation is needed by `ch_ephem.coord.peak_ra`. If we don't want to do this to these `Observers`, one other options would be to keep `peak_ra` in `ch_util` (moving it to, say, `ch_util.tools`), and then there's no need for rotation information in this module.

Like `sources`, this submodule reads data from disk at import time, so is not automatically imported into the base `ch_ephem` object.

## Pointing:

Just Galt telescope pointing routines, unchanged from `ch_util.ephemeris`.  These are another candidate we could consider putting somewhere else in ch_util, if we don't want them here.

## Sources:

Loads the standard source catalogs, and provides the calibrator objects `CasA`, `CygA`, `TauA`, `VirA`.  Like `observers`, this submodule reads data from disk at import time, so is not automatically imported into the base `ch_ephem` object.

## Time:

Time utilities, largely unchanged from what was in `ch_util.ephemeris`.

### CODE REVIEW

**Comment by ketiltrout** [comment]
To pull out one major question I had from the description that I'd like opinions on:

* Should the rotation data be in `ch_ephem`?  Or would it be better to keep it in `ch_util` and move the `peak_ra` function there?  Or just delete `peak_ra` and keep rotation in `ch_util`?

File: .github/workflows/linting.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  pull_request:
+    branches:
+    - master
+  push:
+    branches:
+    - master
+
+jobs:
+
+  lint-code:
```
Comment (ljgray) [inline .github/workflows/linting.yaml]: I wouldn't mind also using the standard ruff config here, but I think that can be done in a different PR

File: ch_ephem/__init__.py [hunk 2]
```diff
@@ -1 +1,30 @@
...
+    pointing
+    sources
+    time
+"""
+
+# We deliberately do not import .observers and .sources here
+# because they both perform reads from disk
+from . import catalogs, coord, pointing, time
+
+from importlib.metadata import version, PackageNotFoundError
```
Comment (ljgray) [inline ch_ephem/__init__.py:96]: We could delete `version` and `PackageNotFoundError` after using them so as not to pollute the namespace. I'm not entirely sure what the accepted best practice is, but I know we do this in some other places

File: ch_ephem/catalogs/__init__.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+import json
+import pathlib
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+del TYPE_CHECKING
+
```
Comment (ljgray) [inline ch_ephem/catalogs/__init__.py:20]: Same comment about namespace

File: ch_ephem/time.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+from caput.time import datetime_to_unix, unix_to_skyfield_time, Observer
+
+
+def parse_date(datestring: str) -> datetime.datetime:
+    """Convert date string to a datetime object.
+
+    Parameters
+    ----------
+    datestring : str
+        Date as YYYYMMDD-AAA, where AAA is one of [UTC, EST, PDT, PST, PDT]
```
Comment (ljgray) [inline ch_ephem/time.py]:
```suggestion
        Date as YYYYMMDD-AAA, where AAA is one of [UTC, EST, EDT, PST, PDT]
```

**Review by ljgray — APPROVED** [review]

**Comment by ljgray** [comment]
> To pull out one major question I had from the description that I'd like opinions on:
> 
> * Should the rotation data be in `ch_ephem`?  Or would it be better to keep it in `ch_util` and move the `peak_ra` function there?  Or just delete `peak_ra` and keep rotation in `ch_util`?

I don't mind moving the rotation data here, personally.

**Comment by ketiltrout** [comment]
I've added `ruff` to the linting (which found all my typos) and move the rest of the geometry from `ch_util.tools` (roll and offset) over here, too.

**Review by ljgray — APPROVED** [review]

**Pull request merged**